File: packages/lotos-dotnet/Contracts/ComponentPropValidator.cs
namespace LotosUi.Dotnet.Contracts;

public sealed class ComponentPropValidator
{
    public IDictionary<string, object?> Validate(string component, IDictionary<string, object?> props)
    {
        if (!ComponentContractRegistry.HasComponent(component))
        {
            throw new ArgumentException($"Unknown LotOS UI component: {component}", nameof(component));
        }

        var allowed = ComponentContractRegistry.AllowedProps(component);
        var sanitized = new Dictionary<string, object?>(StringComparer.OrdinalIgnoreCase);

        foreach (var entry in props)
        {
            if (allowed.Contains(entry.Key))
            {
                sanitized[entry.Key] = entry.Value;
            }
        }

        return sanitized;
    }
}

File: packages/lotos-spring/src/main/java/com/lotosui/spring/contracts/ComponentPropValidator.java
package com.lotosui.spring.contracts;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class ComponentPropValidator {
  public Map<String, Object> validate(String component, Map<String, Object> props) {
    if (!ComponentContractRegistry.hasComponent(component)) {
      throw new IllegalArgumentException("Unknown LotOS UI component: " + component);
    }

    List<String> allowedProps = ComponentContractRegistry.getAllowedProps(component);
    Map<String, Object> sanitized = new HashMap<>();

    for (Map.Entry<String, Object> entry : props.entrySet()) {
      if (allowedProps.contains(entry.getKey())) {
        sanitized.put(entry.getKey(), entry.getValue());
      }
    }

    return sanitized;
  }
}
